qiskit/aqua/algorithms/vq_algorithm.py

```diff
@@ -44,9 +45,10 @@ class VQAlgorithm(QuantumAlgorithm):
     """
     The Variational Quantum Algorithm Base Class.
     """
+
     def __init__(self,
-                 var_form: VariationalForm,
-                 optimizer: Optimizer,
+                 var_form: Optional[VariationalForm] = None,
```

Comment:
In the current structure I don't think it is always possible to provide defaults. E.g. initialize a VQE where the operator is not set. Then you can't set the varform since you don't know the number of qubits. I think it would be more in-line with the late initialization if `VQAlgorithm` already supported it and we don't have to patch support for `None` in the derived classes (since all of them should support it). What do you think?

Reply:
After the Anstaz refactor setting a var form without num qubits will be fine right. The operator in VQE has already been made optional so it can be given later and the ansatz would be updated with the right number of qubits later. I was expecting that derived classes would generate defaults if they allowed None such that ideally this base class never needed to see or deal with None. I do not not have a strong opinion one way or the other on VQ allowing None, I was principally explaining why I had not made them optional and intended to have derived classes build out valid defaults when needed - the agreed intent was to instantiate defaults.

Reply:
Yes that will be fine. But what if the question would rather be what happens if the user doesn't even provide a variational form, nor an Operator? Say
```
vqe = VQE()
```

Reply:
The intent was that VQE (or other algo supporting None) had defaults for Ansatz and Optimizer. Operator will have to be set later which was principally done to support the stack. In allowing Operator to be none, to keep parameter order compatibility var form and optimizer params had to change and defaulted to None similarly with the expectation that we would internally set defaults.
Currently VQE checks if Operator was never set. But more would be needed to allow var form to be none, such as construct circuit check. Since we can set defaults it seemed to be the 'safer' option - but we can allow None too. I just checked later in vq algo find minimum just in case. Before it used to check in its constructor.

Reply:
The `var_form` argument is not optional anymore 👍